### PR TITLE
[do not merge] Old portability collection PR, for reference

### PR DESCRIPTION
**Pull request by rickardp**
This is essentially the same as PR #257. As the old PR was force pushed beyond repair, I'm reposting this for the purpose of referencing the diffs and possibly cherry picking useful parts into other PRs.

The PR sets up a pipeline that builds the .so files separately using a build matrix, then builds binary wheels for the Python packages. The MPS stuff here is just a stub and should not be merged.

AFAIK, the CMake files in this branch contain all the releevant patches to make the CPU code buildable/testable on Windows x64, Mac x64/arm64, Linux x64/arm64.

To summarize this PR should provide

- A CMake file that compiles "eveywhere" and builds one target at a time
- A GitHub workflow that builds first the native libs using a build matrix, then puts everything together in a bdist_wheel
- Source level changes to make native code compile on Windows, Mac and arm64
- MPS stub code for future reference
- Python code refactoring. Probably not worth keeping

### CODE REVIEW

**Comment by Titus-von-Koeller** [comment]
Ok, we'll keep it here for reference. Please close it once not needed anymore. We'll approach you soon about cross-platform support. Thanks for all the good work so far. Let's work together to get this figured out. We're still a bit behind on getting on top of maintenance since we took over, so please excuse the delays. Thanks for your understanding!

**Comment by Titus-von-Koeller** [comment]
Closing this as it seems dead and not needed anymore.

Let me know if it needs reopening.
